### ISSUE TITLE
Feature Request: Scroll to Zoom in 1.12.0-alpha-2

### ISSUE THREAD

**Issue by johannmentz** [camunda/camunda-modeler#738]
I have realised that you have taken the scroll to zoom function away in the alpha version of 1.12. 

This was one of my favourite features, although it took some adjustment coming from modelling on Visio to get used to it, after using the modeller for a few day this really helped me to draw much faster and navigate through my models.

Will you please bring it back 🥇

**Comment by nikku** [comment]
Thanks for your feedback.

There is many good reasons for both ways of interacting with the diagram, _scroll to scroll_ or _scroll to zoom_. We gave _scroll to zoom_ a try for the last years, figured out however that it has major drawbacks when looking into overall tool usage: 

* __Not intuitive to get started.__ As you mention yourself it _took some adjustment_ coming from other tools. As it turns out, it takes this adjustment every day because literally any tool uses _scroll to scroll_. 
* __Does not embed nicely into the web.__ If you think about modeling in Camunda Modeler as well as in the web it is our responsibility to provide a unified way of interacting with diagrams. As a matter of fact _scroll to scroll_ is the standard when navigating web pages, too.
* __No support for native finger gestures.__ Many users nowadays use Notebooks or other trackpad / touch enabled devices. They uses two finger gestures such as pinch-to-zoom. With _scroll to zoom_ we were not able to support these gestures. Now we are.

If you'd like to read more about this, checkout the [release blog post](https://bpmn.io/blog/posts/2018-bpmn-js-0-26.html) of our BPMN modeling toolkit. 

Long story short we'll probably not bring _scroll to zoom_ back. There is a simple workaround though: Use `CTRL + scroll` to zoom (which is the standard in desktop applications and the browser, too).

**Comment by NPDeehan** [comment]
I'd like to add some counter points to this:
 - Not intuitive to get started.
  -- We shouldn't be trying to cater for Viso users, Viso i awful to use when modeling and adding a feature because it makes it easier for people who've gotten use to a worse modeling experience is not a good idea
- Does not embed nicely into the web
 -- If you think about platforms with a large canvas e.g. Google maps where you do a lot of zooming in and out it actaually uses scroll to zoom and works perfectly well. 
- No support for native finger gestures.
 -- the only reason someone will be using a tackpad to model is because they don't have a mouse handy. every aspect of modeling is geared towards precises clicks and dragging items. These are always going to be harder with a trackpad and i don't think we should be accommodating if it sacrifices usability for mouse users.

**Comment by benhoffmann** [comment]
I support the large canvas argument of Niall and I remember you Nico defending scroll-to-zoom back then because of the very same point. All the maps tools do it as well. In order to get from A to B in a large model, I rather zoom out and back into it instead of scrolling all the way.

That's why I strongly opt for not changing it. I guess it will drive people who are used to using the modeler for a while absolutely nuts if you change it (it does with me). 

Why not make it configurable and let people decide on their own? That would be a good compromise.

Also, there is a native gesture (at least on Mac touchpads) for panning large canvases and that is three fingers, which works absolutely fine.

**Comment by nikku** [comment]
> Also, there is a native gesture (at least on Mac touchpads) for panning large canvases and that is three fingers, which works absolutely fine. AND: you already support pinch-to-zoom with two fingers as well ;-)

Did that work with Camunda Modeler v1.11.x?

**Comment by benhoffmann** [comment]
Yeah, I use it all the time. Maybe it's a Mac-specific electron thing, I don't know. But both scroll-to-zoom and pinch-to-zoom work with two fingers and panning works with three fingers since quite a while I think.

**Comment by nikku** [comment]
Link to the [original feature request](https://github.com/camunda/camunda-modeler/issues/629).

**Comment by benhoffmann** [comment]
Hm... this seems to be a controversial topic. 

Would it be that hard to provide an option and let the user decide?

**Comment by nevries** [comment]
Ok, next guy giving his 50c.

The option thing is semi-optimal in my opinion since we not only have the standalone modeler but also web applications, embedding in web-pages,  etc.
And, the option exists by pressing CTRL.

One thing that drives newcomers nuts (and it does with me) is that there is no proper gesture for panning. Especially on large diagrams with pools it is too difficult to pan. You have to use hand tool or explicitly move the pointer to a portion of the canvas not covered by pool and then use dragging (this is also true for current 3-finger-panning).
Solving this counts for me more than habit arguments or comparisons to large canvas web-apps like maps (which, btw, also behave different when embedded in smaller canvases -- which is terrible imo).

> -- the only reason someone will be using a tackpad to model is because they don't have a mouse handy.

Modelling on good trackpads works like a charm!

**Comment by benhoffmann** [comment]
> One thing that drives newcomers nuts (and it does with me) is that there is no proper gesture for panning. Especially on large diagrams with pools it is too difficult to pan.

Never tried it myself with the modeler, but clicking the mouse wheel should be the equivalent of the mac three finger pan, no?

> move the pointer to a portion of the canvas not covered by pool and then use dragging (this is also true for current 3-finger-panning).

Yeah, I think here lies really the most painful experience for newcomers, panning when the cursor is inside pools. Friendly reminder: https://github.com/camunda/camunda-modeler/issues/238

**Comment by benhoffmann** [comment]
One more comment:

> Use CTRL + scroll to zoom (which is the standard in desktop applications and the browser, too).

Please make it at least Cmd + scroll for Mac users if you decide to keep it.

**Comment by nikku** [comment]
> One thing that drives newcomers nuts (and it does with me) is that there is no proper gesture for panning.  

The recommended gesture for panning is `CTRL + Drag`.
